src/Drawables/Drawable.vala

```diff
@@ -228,6 +228,55 @@ public class Akira.Drawables.Drawable {
         context.new_path ();
     }
 
+    /*
+     * Create a target like element on top of the drawable to represent the
+     * current subselection anchor point that will be used as alignment pivot.
+     */
+    public virtual void paint_anchor (
+        Cairo.Context context,
+        Gdk.RGBA color,
+        double line_width,
+        double scale
+    ) {
+        context.save ();
+        Cairo.Matrix global_transform = context.get_matrix ();
+
+        // We apply the item transform before creating the path
+        Cairo.Matrix tr = transform;
+        context.transform (tr);
+
+        // Set the visual properties.
+        context.set_line_width (line_width / scale);
+        context.set_source_rgba (color.red, color.green, color.blue, color.alpha);
+
+        // Create the circle.
+        context.new_path ();
+        context.arc (0.0, 0.0, 5 / scale, 0.0, 2.0 * GLib.Math.PI);
+        // Draw the circle and preserve the visual properties.
+        context.stroke_preserve ();
+
+        var lenght = 10 / scale;
+        // Create the horizontal stroke.
+        context.new_path ();
+        context.move_to (-lenght, 0);
+        context.line_to (lenght, 0);
+        context.stroke_preserve ();
```

Comment:
why are you using stroke_preserve on these calls? you are re-painting each path every time you do stroke